shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -444,11 +454,17 @@ void onDestroyView() {
   }
 
   void onSaveInstanceState(@Nullable Bundle bundle) {
-    Log.v(TAG, "onSaveInstanceState. Giving plugins an opportunity to save state.");
+    Log.v(TAG, "onSaveInstanceState. Giving framework and plugins an opportunity to save state.");
     ensureAlive();
 
+    bundle.putByteArray(
+        FRAMEWORK_BUNDLE_KEY,
+        flutterEngine.getRestorationChannel().getRestorationDataFromFramework());
```

Comment:
theoretically the channels (via the engine) and consequently those byte arrays can live forever. Don't we want to get rid of them at some point?

In case the FlutterActivity keeps dying but the owner of the FlutterEngine never does.

Reply:
The channels and those byte arrays have the same life time as the engine. The byte arrays represent the current state of the framework running in a given engine. So, tying their life cycle to the lifecycle of the engine seems appropriate to me.